src/proxy.sol

```diff
@@ -50,19 +50,26 @@ contract DSProxy is DSAuth, DSNote {
         response = execute(target, _data);
     }
 
+
     function execute(address _target, bytes _data)
         public
         auth
         note
         payable
-        returns (bytes32 response)
+        returns (bytes response)
     {
         require(_target != 0x0);
 
         // call contract in current context
         assembly {
-            let succeeded := delegatecall(sub(gas, 5000), _target, add(_data, 0x20), mload(_data), 0, 32)
-            response := mload(0)      // load delegatecall output
+            let succeeded := delegatecall(sub(gas, 5000), _target, add(_data, 0x20), mload(_data), 0, 0)
+            let size := returndatasize
+
+            response := mload(0x40)
+            mstore(0x40, add(response, and(add(add(size, 0x20), 0x1f), not(0x1f))))
+            mstore(response, size)
+            returndatacopy(add(response, 0x20), 0, size)
+
             switch iszero(succeeded)
             case 1 {
                 // throw if delegatecall failed
```

Comment:
this `revert(0, 0)` should change to `revert(0, response)` so that revert messages are propagated up the call chain

Reply:
made #19 for this